flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/DataMapManager.java

```diff
@@ -525,27 +535,52 @@ private void genMapping(ListConstructorExpressionNode listCtrExpr, List<Mapping>
         mappings.add(mapping);
     }
 
-    private void genMapping(Node expr, String name, List<Mapping> elements, SemanticModel semanticModel) {
+    private void genMapping(Node expr, String name, List<Mapping> elements, SemanticModel semanticModel,
+                            Document functionDocument) {
         List<String> inputs = new ArrayList<>();
         genInputs(expr, inputs);
+        LineRange customFunctionRange = getCustomFunctionRange(expr, functionDocument);
         Mapping mapping = new Mapping(name, inputs, expr.toSourceCode(),
                 getDiagnostics(expr.lineRange(), semanticModel), new ArrayList<>(),
-                expr.kind() == SyntaxKind.QUERY_EXPRESSION);
+                expr.kind() == SyntaxKind.QUERY_EXPRESSION, customFunctionRange);
         elements.add(mapping);
     }
 
+    private LineRange getCustomFunctionRange(Node expr, Document functionDocument) {
+        if (functionDocument == null) {
+            return null;
+        }
+        if (expr.kind() != SyntaxKind.FUNCTION_CALL) {
+            return null;
+        }
```

Comment:
We can combine these two into a one `if` condition with `OR`

Reply:
Fixed